colcon_output/event_handler/console_package_list.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2016-2018 Dirk Thomas
+# Licensed under the Apache License, Version 2.0
+
+from colcon_core.event.job import JobQueued
+from colcon_core.event.job import JobStarted
+from colcon_core.event_handler import EventHandlerExtensionPoint
+from colcon_core.plugin_system import satisfies_version
+
+
+class ConsolePackageListEventHandler(EventHandlerExtensionPoint):
+    """
+    Output list of queued job names.
+
+    The extension handles events of the following types:
+    - :py:class:`colcon_core.event.job.JobQueued`
+    - :py:class:`colcon_core.event.job.JobStarted`
+    """
+
+    def __init__(self):  # noqa: D107
+        super().__init__()
+        satisfies_version(
+            EventHandlerExtensionPoint.EXTENSION_POINT_VERSION, '^1.0')
+        self.enabled = False
+        self._queued = []
+
+    def __call__(self, event):  # noqa: D102
+        data = event[0]
+
+        if isinstance(data, JobQueued):
+            job = event[1]
+            self._queued.append(job)
+        if isinstance(data, JobStarted):
+            if self._queued:
+                print('Topological order')
+                for job in self._queued:
+                    print(
+                        '- {job} ({job._task_context.pkg.type})'
```

Comment:
Before merging this I would expose `_task_context` in the `Job` class in order to not rely on access to "protected" fields.

Reply:
ffffd623a7f087ae4701ccc9b43c5e7ee732bf0f relies on colcon/colcon-core#96.